LAPACK bug is breaking the build of 0.3.30 now: `collect2: fatal error: posix_spawnp: Argument list too long`

@martin-frbg Now that LAPACK bug propagated into the release here:
```
[ 82%] Building C object CMakeFiles/LAPACKE.dir/lapack-netlib/LAPACKE/utils/lapacke_stz_trans.c.o
/opt/local/bin/gcc-mp-14  -I/opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/OpenBLAS-0.3.30/lapack-netlib/LAPACKE/include -pipe -O3 -DNDEBUG -I/opt/local/include -arch ppc  -DHAVE_C11 -Wall -m32 -mtune=G4 -maltivec -fno-fast-math -force_cpusubtype_ALL -mtune=G4 -maltivec -fno-fast-math -force_cpusubtype_ALL -fPIC -DNO_AVX -DNO_AVX2 -DNO_AVX512 -DSMP_SERVER -DNO_WARMUP -DMAX_CPU_NUMBER=56 -DMAX_PARALLEL_NUMBER=1 -DMAX_STACK_ALLOC=2048 -DNO_AFFINITY -DVERSION="\"0.3.30\"" -DBUILD_SINGLE -DBUILD_DOUBLE -DBUILD_COMPLEX -DBUILD_COMPLEX16 -arch ppc -mmacosx-version-min=10.6  -DHAVE_LAPACK_CONFIG_H -MD -MT CMakeFiles/LAPACKE.dir/lapack-netlib/LAPACKE/utils/lapacke_stz_trans.c.o -MF CMakeFiles/LAPACKE.dir/lapack-netlib/LAPACKE/utils/lapacke_stz_trans.c.o.d -o CMakeFiles/LAPACKE.dir/lapack-netlib/LAPACKE/utils/lapacke_stz_trans.c.o -c /opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/OpenBLAS-0.3.30/lapack-netlib/LAPACKE/utils/lapacke_stz_trans.c
[ 82%] Building C object CMakeFiles/LAPACKE.dir/lapack-netlib/LAPACKE/utils/lapacke_stz_nancheck.c.o
/opt/local/bin/gcc-mp-14  -I/opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/OpenBLAS-0.3.30/lapack-netlib/LAPACKE/include -pipe -O3 -DNDEBUG -I/opt/local/include -arch ppc  -DHAVE_C11 -Wall -m32 -mtune=G4 -maltivec -fno-fast-math -force_cpusubtype_ALL -mtune=G4 -maltivec -fno-fast-math -force_cpusubtype_ALL -fPIC -DNO_AVX -DNO_AVX2 -DNO_AVX512 -DSMP_SERVER -DNO_WARMUP -DMAX_CPU_NUMBER=56 -DMAX_PARALLEL_NUMBER=1 -DMAX_STACK_ALLOC=2048 -DNO_AFFINITY -DVERSION="\"0.3.30\"" -DBUILD_SINGLE -DBUILD_DOUBLE -DBUILD_COMPLEX -DBUILD_COMPLEX16 -arch ppc -mmacosx-version-min=10.6  -DHAVE_LAPACK_CONFIG_H -MD -MT CMakeFiles/LAPACKE.dir/lapack-netlib/LAPACKE/utils/lapacke_stz_nancheck.c.o -MF CMakeFiles/LAPACKE.dir/lapack-netlib/LAPACKE/utils/lapacke_stz_nancheck.c.o.d -o CMakeFiles/LAPACKE.dir/lapack-netlib/LAPACKE/utils/lapacke_stz_nancheck.c.o -c /opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/OpenBLAS-0.3.30/lapack-netlib/LAPACKE/utils/lapacke_stz_nancheck.c
[ 82%] Building C object CMakeFiles/LAPACKE.dir/lapack-netlib/LAPACKE/utils/lapacke_ztz_nancheck.c.o
/opt/local/bin/gcc-mp-14  -I/opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/OpenBLAS-0.3.30/lapack-netlib/LAPACKE/include -pipe -O3 -DNDEBUG -I/opt/local/include -arch ppc  -DHAVE_C11 -Wall -m32 -mtune=G4 -maltivec -fno-fast-math -force_cpusubtype_ALL -mtune=G4 -maltivec -fno-fast-math -force_cpusubtype_ALL -fPIC -DNO_AVX -DNO_AVX2 -DNO_AVX512 -DSMP_SERVER -DNO_WARMUP -DMAX_CPU_NUMBER=56 -DMAX_PARALLEL_NUMBER=1 -DMAX_STACK_ALLOC=2048 -DNO_AFFINITY -DVERSION="\"0.3.30\"" -DBUILD_SINGLE -DBUILD_DOUBLE -DBUILD_COMPLEX -DBUILD_COMPLEX16 -arch ppc -mmacosx-version-min=10.6  -DHAVE_LAPACK_CONFIG_H -MD -MT CMakeFiles/LAPACKE.dir/lapack-netlib/LAPACKE/utils/lapacke_ztz_nancheck.c.o -MF CMakeFiles/LAPACKE.dir/lapack-netlib/LAPACKE/utils/lapacke_ztz_nancheck.c.o.d -o CMakeFiles/LAPACKE.dir/lapack-netlib/LAPACKE/utils/lapacke_ztz_nancheck.c.o -c /opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/OpenBLAS-0.3.30/lapack-netlib/LAPACKE/utils/lapacke_ztz_nancheck.c
[ 82%] Building C object CMakeFiles/LAPACKE.dir/lapack-netlib/LAPACKE/utils/lapacke_ztz_trans.c.o
/opt/local/bin/gcc-mp-14  -I/opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/OpenBLAS-0.3.30/lapack-netlib/LAPACKE/include -pipe -O3 -DNDEBUG -I/opt/local/include -arch ppc  -DHAVE_C11 -Wall -m32 -mtune=G4 -maltivec -fno-fast-math -force_cpusubtype_ALL -mtune=G4 -maltivec -fno-fast-math -force_cpusubtype_ALL -fPIC -DNO_AVX -DNO_AVX2 -DNO_AVX512 -DSMP_SERVER -DNO_WARMUP -DMAX_CPU_NUMBER=56 -DMAX_PARALLEL_NUMBER=1 -DMAX_STACK_ALLOC=2048 -DNO_AFFINITY -DVERSION="\"0.3.30\"" -DBUILD_SINGLE -DBUILD_DOUBLE -DBUILD_COMPLEX -DBUILD_COMPLEX16 -arch ppc -mmacosx-version-min=10.6  -DHAVE_LAPACK_CONFIG_H -MD -MT CMakeFiles/LAPACKE.dir/lapack-netlib/LAPACKE/utils/lapacke_ztz_trans.c.o -MF CMakeFiles/LAPACKE.dir/lapack-netlib/LAPACKE/utils/lapacke_ztz_trans.c.o.d -o CMakeFiles/LAPACKE.dir/lapack-netlib/LAPACKE/utils/lapacke_ztz_trans.c.o -c /opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/OpenBLAS-0.3.30/lapack-netlib/LAPACKE/utils/lapacke_ztz_trans.c
make[2]: Leaving directory `/opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/build'
[ 82%] Built target LAPACKE
/usr/bin/make  -f CMakeFiles/openblas_shared.dir/build.make CMakeFiles/openblas_shared.dir/depend
make[2]: Entering directory `/opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/build'
cd /opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/build && /opt/local/bin/cmake -E cmake_depends "Unix Makefiles" /opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/OpenBLAS-0.3.30 /opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/OpenBLAS-0.3.30 /opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/build /opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/build /opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/build/CMakeFiles/openblas_shared.dir/DependInfo.cmake "--color="
Dependee "/opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/build/CMakeFiles/openblas_shared.dir/DependInfo.cmake" is newer than depender "/opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/build/CMakeFiles/openblas_shared.dir/depend.internal".
Dependee "/opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/build/CMakeFiles/CMakeDirectoryInformation.cmake" is newer than depender "/opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/build/CMakeFiles/openblas_shared.dir/depend.internal".
Scanning dependencies of target openblas_shared
make[2]: Leaving directory `/opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/build'
/usr/bin/make  -f CMakeFiles/openblas_shared.dir/build.make CMakeFiles/openblas_shared.dir/build
make[2]: Entering directory `/opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/build'
[ 82%] Linking Fortran shared library lib/libopenblas.dylib
/opt/local/bin/cmake -E cmake_link_script CMakeFiles/openblas_shared.dir/link.txt --verbose=ON
collect2: fatal error: posix_spawnp: Argument list too long
compilation terminated.
/opt/local/bin/gfortran-mp-14 -pipe -Os -m32  -fPIC -mmacosx-version-min=10.6 -dynamiclib -Wl,-headerpad_max_install_names -L/opt/local/lib -Wl,-headerpad_max_install_names -arch ppc -current_version 0.3.0 -o lib/libopenblas.0.3.dylib -install_name @rpath/libopenblas.0.dylib @CMakeFiles/openblas_shared.dir/objects1.rsp @CMakeFiles/openblas_shared.dir/objects2.rsp @CMakeFiles/openblas_shared.dir/objects3.rsp @CMakeFiles/openblas_shared.dir/objects4.rsp  -lm
make[2]: *** [lib/libopenblas.0.3.dylib] Error 1
```
Could we do something about that? I will do testing, of course.

After removing DYNAMIC_ARCH condition, so that the hack is used, I got a
segfault:
```
[ 83%] Building C object
utest/CMakeFiles/openblas_utest_ext.dir/test_extensions/test_scsum.c.o
cd
/opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/build/utest
&& /opt/local/bin/gcc-mp-14
-I/opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/OpenBLAS-0.3.30/lapack-netlib/LAPACKE/include
-I/opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/OpenBLAS-0.3.30
-I/opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/build
-pipe -O3 -DNDEBUG -I/opt/local/include -arch ppc  -DHAVE_C11 -Wall -m32
-mtune=G4 -maltivec -fno-fast-math -force_cpusubtype_ALL -mtune=G4
-maltivec -fno-fast-math -force_cpusubtype_ALL -fPIC -DNO_AVX -DNO_AVX2
-DNO_AVX512 -DSMP_SERVER -DNO_WARMUP -DMAX_CPU_NUMBER=56
-DMAX_PARALLEL_NUMBER=1 -DMAX_STACK_ALLOC=2048 -DNO_AFFINITY
-DVERSION="\"0.3.30\"" -DBUILD_SINGLE -DBUILD_DOUBLE -DBUILD_COMPLEX
-DBUILD_COMPLEX16 -arch ppc -mmacosx-version-min=10.6 -MD -MT
utest/CMakeFiles/openblas_utest_ext.dir/test_extensions/test_scsum.c.o -MF
CMakeFiles/openblas_utest_ext.dir/test_extensions/test_scsum.c.o.d -o
CMakeFiles/openblas_utest_ext.dir/test_extensions/test_scsum.c.o -c
/opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/OpenBLAS-0.3.30/utest/test_extensions/test_scsum.c
Dependee
"/opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/build/test/CMakeFiles/cblat3.dir/DependInfo.cmake"
is newer than depender
"/opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/build/test/CMakeFiles/cblat3.dir/depend.internal".
Dependee
"/opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/build/test/CMakeFiles/CMakeDirectoryInformation.cmake"
is newer than depender
"/opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/build/test/CMakeFiles/cblat3.dir/depend.internal".
Scanning dependencies of target cblat3
make[2]: Leaving directory
`/opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/build'
/usr/bin/make  -f test/CMakeFiles/cblat3.dir/build.make
test/CMakeFiles/cblat3.dir/build
make[2]: Entering directory
`/opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/build'
[ 83%] Linking C executable openblas_utest
cd
/opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/build/utest
&& /opt/local/bin/cmake -E cmake_link_script
CMakeFiles/openblas_utest.dir/link.txt --verbose=ON
[ 83%] Building C object
utest/CMakeFiles/openblas_utest_ext.dir/test_extensions/test_dzsum.c.o
cd
/opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/build/utest
&& /opt/local/bin/gcc-mp-14
-I/opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/OpenBLAS-0.3.30/lapack-netlib/LAPACKE/include
-I/opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/OpenBLAS-0.3.30
-I/opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/build
-pipe -O3 -DNDEBUG -I/opt/local/include -arch ppc  -DHAVE_C11 -Wall -m32
-mtune=G4 -maltivec -fno-fast-math -force_cpusubtype_ALL -mtune=G4
-maltivec -fno-fast-math -force_cpusubtype_ALL -fPIC -DNO_AVX -DNO_AVX2
-DNO_AVX512 -DSMP_SERVER -DNO_WARMUP -DMAX_CPU_NUMBER=56
-DMAX_PARALLEL_NUMBER=1 -DMAX_STACK_ALLOC=2048 -DNO_AFFINITY
-DVERSION="\"0.3.30\"" -DBUILD_SINGLE -DBUILD_DOUBLE -DBUILD_COMPLEX
-DBUILD_COMPLEX16 -arch ppc -mmacosx-version-min=10.6 -MD -MT
utest/CMakeFiles/openblas_utest_ext.dir/test_extensions/test_dzsum.c.o -MF
CMakeFiles/openblas_utest_ext.dir/test_extensions/test_dzsum.c.o.d -o
CMakeFiles/openblas_utest_ext.dir/test_extensions/test_dzsum.c.o -c
/opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/OpenBLAS-0.3.30/utest/test_extensions/test_dzsum.c
[ 83%] Building Fortran object test/CMakeFiles/cblat3.dir/cblat3.f.o
cd
/opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/build/test
&& /opt/local/bin/gfortran-mp-14
-I/opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/OpenBLAS-0.3.30/lapack-netlib/LAPACKE/include
-I/opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/OpenBLAS-0.3.30
-I/opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/build
-pipe -Os -m32  -fPIC -fno-tree-vectorize -mmacosx-version-min=10.6 -c
/opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/OpenBLAS-0.3.30/test/cblat3.f
-o CMakeFiles/cblat3.dir/cblat3.f.o
[ 83%] Building C object
utest/CMakeFiles/openblas_utest_ext.dir/test_extensions/test_samin.c.o
cd
/opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/build/utest
&& /opt/local/bin/gcc-mp-14
-I/opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/OpenBLAS-0.3.30/lapack-netlib/LAPACKE/include
-I/opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/OpenBLAS-0.3.30
-I/opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/build
-pipe -O3 -DNDEBUG -I/opt/local/include -arch ppc  -DHAVE_C11 -Wall -m32
-mtune=G4 -maltivec -fno-fast-math -force_cpusubtype_ALL -mtune=G4
-maltivec -fno-fast-math -force_cpusubtype_ALL -fPIC -DNO_AVX -DNO_AVX2
-DNO_AVX512 -DSMP_SERVER -DNO_WARMUP -DMAX_CPU_NUMBER=56
-DMAX_PARALLEL_NUMBER=1 -DMAX_STACK_ALLOC=2048 -DNO_AFFINITY
-DVERSION="\"0.3.30\"" -DBUILD_SINGLE -DBUILD_DOUBLE -DBUILD_COMPLEX
-DBUILD_COMPLEX16 -arch ppc -mmacosx-version-min=10.6 -MD -MT
utest/CMakeFiles/openblas_utest_ext.dir/test_extensions/test_samin.c.o -MF
CMakeFiles/openblas_utest_ext.dir/test_extensions/test_samin.c.o.d -o
CMakeFiles/openblas_utest_ext.dir/test_extensions/test_samin.c.o -c
/opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/OpenBLAS-0.3.30/utest/test_extensions/test_samin.c
collect2: fatal error: ld terminated with signal 11 [Segmentation fault]
compilation terminated.
/opt/local/bin/gcc-mp-14 -pipe -O3 -DNDEBUG -I/opt/local/include -arch ppc
-DHAVE_C11 -Wall -m32 -mtune=G4 -maltivec -fno-fast-math
-force_cpusubtype_ALL -mtune=G4 -maltivec -fno-fast-math
-force_cpusubtype_ALL -fPIC -DNO_AVX -DNO_AVX2 -DNO_AVX512 -DSMP_SERVER
-DNO_WARMUP -DMAX_CPU_NUMBER=56 -DMAX_PARALLEL_NUMBER=1
-DMAX_STACK_ALLOC=2048 -DNO_AFFINITY -DVERSION="\"0.3.30\"" -DBUILD_SINGLE
-DBUILD_DOUBLE -DBUILD_COMPLEX -DBUILD_COMPLEX16 -arch ppc
-mmacosx-version-min=10.6 -Wl,-search_paths_first
-Wl,-headerpad_max_install_names -L/opt/local/lib
-Wl,-headerpad_max_install_names -arch ppc
@CMakeFiles/openblas_utest.dir/objects1.rsp -o openblas_utest
-Wl,-rpath,/opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/build/lib
../lib/libopenblas.0.3.dylib -lm
make[2]: *** [utest/openblas_utest] Error 1
make[2]: *** Deleting file `utest/openblas_utest'
make[2]: Leaving directory
`/opt/local/var/macports/build/_opt_local_ppcports_math_OpenBLAS/OpenBLAS/work/build'
make[1]: *** [utest/CMakeFiles/openblas_utest.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

On Mon, Jun 23, 2025 at 3:53 PM Martin Kroeker ***@***.***>
wrote:

> *martin-frbg* left a comment (OpenMathLib/OpenBLAS#5332)
> <https://github.com/OpenMathLib/OpenBLAS/issues/5332#issuecomment-2995334130>
>
> 0.3.30 contains the supposed fix for #5223
> <https://github.com/OpenMathLib/OpenBLAS/issues/5223>, so if that doesn't
> help I'm not sure what to suggest, except perhaps see if you can run the
> build in /tmp so that object paths take up fewer characters
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/OpenMathLib/OpenBLAS/issues/5332#issuecomment-2995334130>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AV6AXFWPIVDD2URL3AUSBVL3E6W7RAVCNFSM6AAAAAB74SOWUOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDSOJVGMZTIMJTGA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -305,8 +305,8 @@ if (USE_OPENMP)
   endif()
 endif()
 
-# Fix "Argument list too long" for macOS with Intel CPUs and DYNAMIC_ARCH turned on
-if(APPLE AND DYNAMIC_ARCH AND (NOT CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL "arm64"))
+# Fix "Argument list too long" for macOS with POWERPC or Intel CPUs 
+if(APPLE AND (NOT CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL "arm64"))
   # Use response files
   set(CMAKE_C_USE_RESPONSE_FILE_FOR_OBJECTS 1)
   # Always build static library first

